src/ORM/Table.php

```diff
@@ -1857,6 +1862,53 @@ public function exists(QueryExpression|Closure|array|string|null $conditions): b
         );
     }
 
+    /**
+     * Helper method to infer the requested finder and its options.
+     *
+     * Returns the inferred options from the finder $finderData.
+     *
+     * ### Examples:
+     *
+     * Given you're using the Muffin/TrashBehavior
+     * The following will call the finder 'withTrashed' with the value of the finder as its options:
+     * ```
+     * $table->Articles->exists(['id' => 1], 'withTrashed');
+     * $table->Articles->exists(['id' => 1], ['withTrashed' => []]);
+     * //this is the same as
+     * $table->Articles->exists(['id' => 1], ['all' => ['skipAddTrashCondition' => true]]);
+     * ```
+     *
+     * Only return true if an article with `en` and `es` locales exist
+     * ```
+     * $table->Articles->exists(['id' => 1], ['translations' => ['locales' => ['en', 'es']]]
+     * ```
+     *
+     * The following will call the finder 'published' with additional options. Those options will be available
+     * inside the attached behaviors (resp. their beforeFind-events):
+     * ```
+     * $table->Articles->exists(['id' => 1], [
+     *  'published' => [
+     *      'published_before' => '2010-01-01 00:00:00',
+     *      'skipAddTrashCondition' => true,
+     *  ]
+     * ]);
+     * ```
+     *
+     * @param array|string $finderData The finder name or an array having the name as key
+     * and options as value.
+     * @return array
+     */
+    protected function _extractFinder(array|string $finderData): array
```

Comment:
Don't we have a similar function somewhere else already?

Reply:
Yes indeed:

https://github.com/cakephp/cakephp/blob/e28e6125405bcf447d5c7120b3af8f0f4f47feff/src/ORM/Association.php#L1147-L1156

https://github.com/cakephp/cakephp/blob/e28e6125405bcf447d5c7120b3af8f0f4f47feff/src/ORM/Association/Loader/SelectLoader.php#L224-L233

I can remove the one from the `Association` and make the one in `Table` public. Not so sure about the one in SelectLoader.